### PR TITLE
[FIXED JENKINS-26051] withCredentials step

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-26051](https://issues.jenkins-ci.org/browse/JENKINS-26051)

- [X] basic impl
- [X] simplify SPI
- [X] add automated tests
- [x] run manual testing

### CODE REVIEW

**Comment by albers** [comment]
@jglick sorry, I accidentally clicked a checkbox in the issue description, not sure about original values, please verify.

**Comment by jglick** [comment]
@albers no problem, I rechecked it.

@reviewbybees

File: src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java [hunk 1]
```diff
@@ -60,21 +63,47 @@ public final String getCredentialsId() {
...
+        }
+
+    }
+
+    /** Callback run at the end of a build. */
+    public interface Unbinder extends Serializable {
         /** Performs any needed cleanup. */
-        void unbind() throws IOException, InterruptedException;
+        void unbind(@Nonnull Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException;
+    }
```
Comment (recampbell) [inline src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java]: I suppose this is an internal interface of which there are no external implementations?

**Comment by rsandell** [comment]
+1

File: src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java]: While I know of no external implementations, this is intended as a _potential_ external interface. At any rate, `MultiBinding` itself was introduced during unreleased changes in #2, so changing it before release cannot be breaking anyone.

**Pull request merged**